components/draw/AttributeMode.svelte

```diff
@@ -0,0 +1,91 @@
+<script>
+  import geojsonExtent from "@mapbox/geojson-extent";
+  import {
+    map,
+    gjScheme,
+    currentHover,
+    currentlyEditing,
+    setCurrentlyEditing,
+    openFromSidebar,
+  } from "../../stores.js";
+
+  export let mode;
+  export let drawControls;
+  // Maybe null at first
+  export let routeSnapper;
+
+  $: {
+    if (mode == "edit-attribute") {
+      drawControls.changeMode("static");
```

Comment:
I'm starting to dislike these reactive statements when `mode` changes, but I'm not sure what else to do. All of the different modes effectively need to do something when they become active, and clean up when they become inactive. There are many ways they can become inactive -- for example, click the route tool to start it, but then click something in the sidebar or change to "Edit attributes" mode. The problem with reactive statements is that there's no ordering guarantees at all, so if multiple things are listening to `mode` change, we don't really know what order stuff is happening. Any ideas for detangling this mess?